### PR TITLE
fix: use record type and add missing properties to AXIS_PROPERTY_TYPE

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/axis.ts [hunk 1]
```diff
@@ -353,10 +361,12 @@ export const AXIS_PROPERTY_TYPE: {
   titleX: 'main',
   titleY: 'main',
 
-  tickWidth: 'both',
+  encode: 'both', // we hide this in Vega-Lite
+  scale: 'both',
   tickCount: 'both',
+  tickWidth: 'both',
```
Comment (kanitw) [inline src/axis.ts]: why both for `tickWidth`?

File: src/axis.ts [hunk 2]
```diff
@@ -321,30 +320,39 @@ export const AXIS_PROPERTY_TYPE: {
...
   labelOpacity: 'main',
   labelOverlap: 'main',
   labelPadding: 'main',
   labels: 'main',
+  labelSeparation: 'main',
   maxExtent: 'main',
   minExtent: 'main',
   offset: 'both',
   position: 'main',
+  tickBand: 'main',
```
Comment (kanitw) [inline src/axis.ts]: tickBand is a shorthand for bandPosition+tickExtra - it should be "both".

File: src/axis.ts [hunk 3]
```diff
@@ -321,30 +320,39 @@ export const AXIS_PROPERTY_TYPE: {
...
   tickColor: 'main',
+  tickDash: 'main',
+  tickDashOffset: 'main',
   tickExtra: 'main',
```
Comment (kanitw) [inline src/axis.ts]: tickExtra actually should be "both".

**Comment by domoritz** [comment]
@kanitw updated

**Pull request merged**